internal/powerbi/resource_pbix.go

```diff
@@ -237,14 +259,14 @@ func deletePBIX(d *schema.ResourceData, meta interface{}) error {
 
 	groupID := d.Get("workspace_id").(string)
 
-	if reportID := d.Get("report_id"); reportID != nil && reportID != "" {
+	if reportID, reportIDOk := d.GetOk("report_id"); reportIDOk {
 		err := client.DeleteReportInGroup(groupID, reportID.(string))
 		if err != nil {
 			return err
 		}
 	}
 
-	if datasetID := d.Get("dataset_id"); datasetID != nil && datasetID != "" {
+	if datasetID, datasetIDOk := d.GetOk("dataset_id"); datasetIDOk {
```

Comment:
Do we need to check something like datasetID! = rebindDatasetId here, as when the dataset has been rebound it should be managed from another resource?